language/libra-vm/src/data_cache.rs

```diff
@@ -75,8 +82,38 @@ impl<'a> StateViewCache<'a> {
 }
 
 impl<'block> RemoteCache for StateViewCache<'block> {
-    fn get(&self, access_path: &AccessPath) -> VMResult<Option<Vec<u8>>> {
-        StateViewCache::get(self, access_path)
+    fn get_module(&self, module_id: &ModuleId) -> VMResult<Option<CompiledModule>> {
```

Comment:
I am not sure I like this more than what it was but I am fine changing it and going out either way.
The thing I like about returning the `Vector<u8>` is that the VM materializes things. The VM is in charge of everything and the only thing in are blobs.
I understand the tension with serialization and access path, and I am not sure we are in a position to make a final call there. In that respect anything works. 
But I invite you to quickly evaluate if this change adds much